### ISSUE TITLE
File rename broken for changing capitalization

### ISSUE THREAD

**Issue by csdrane** [syl20bnr/spacemacs#4383]
I'm running 0.105.0 but this was an issue on earlier versions as well. Renaming `./Foo.md` to `./FOO.md` using `SPC f R` fails (I receive a prompt telling me I'm attempting to rename Foo.md to Foo.md). I suspect that this is due to the autocomplete changing FOO back to Foo without showing so to the user. I didn't see any way to disable this behavior to test this.

**Comment by TheBB** [comment]
The helm prompt should have two lines, one that says `Foo.md` and another that says `[?] FOO.md`. The latter shows what you typed, literally. You have to choose that one.

**Comment by csdrane** [comment]
The `FOO.md` line is in fact selected by default. Selecting this line elicits the problem above.

**Comment by TheBB** [comment]
Okay. I can't reproduce then. I can successfully rename Foo to FOO.

**Comment by csdrane** [comment]
Is there any pertinent debugging info that I can provide? This bug is persistent across directories and versions.

**Comment by StreakyCobra** [comment]
You can paste here the output of `SPC h d s`.

**Comment by csdrane** [comment]
#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: master (rev. d2d53f5)
- Distribution: spacemacs
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-help-tooltip t auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
 (c-c++ :variables c-c++-enable_clang-support t)
 clojure emacs-lisp extra-langs erc eyebrowse git github haskell markdown org spacemacs-layouts version-control)
```

**Comment by TheBB** [comment]
I thought OSX filesystems weren't case sensitive?

**Comment by csdrane** [comment]
That's correct. I don't see why that should prevent me from renaming a file though.

**Comment by StreakyCobra** [comment]
What the value of `helm-file-name-case-fold-search` (`SPC h d v helm-file-name-case-fold-search RET`) ?

**Comment by csdrane** [comment]
Set to `t`

**Comment by StreakyCobra** [comment]
For me it's set to `smart`.

**Comment by nixmaniack** [comment]
Not sure of exact issue(on osx as well here), I tried setting `read-file-name-completion-ignore-case`, `helm-file-name-case-fold-search` and `helm-comp-read-case-fold-search` to `nil` without success. It looks like passing correct dir value to [`read-file-name`](https://github.com/syl20bnr/spacemacs/blob/02f7500c867fc14c021769fe7f12459f275e9e20/layers/%2Bdistribution/spacemacs-base/funcs.el#L285) allows me to rename successfully. 

```diff
diff --git a/layers/+distribution/spacemacs-base/funcs.el b/layers/+distribution/spacemacs-base/funcs.el
index 483c143..08af315 100644
--- a/layers/+distribution/spacemacs-base/funcs.el
+++ b/layers/+distribution/spacemacs-base/funcs.el
@@ -278,11 +278,12 @@ argument takes the kindows rotate backwards."
 (defun spacemacs/rename-current-buffer-file ()
   "Renames current buffer and file it is visiting."
   (interactive)
-  (let ((name (buffer-name))
-        (filename (buffer-file-name)))
+  (let* ((name (buffer-name))
+         (filename (buffer-file-name))
+         (dir (file-name-directory filename)))
     (if (not (and filename (file-exists-p filename)))
         (error "Buffer '%s' is not visiting a file!" name)
-      (let ((new-name (read-file-name "New name: " filename)))
+      (let ((new-name (read-file-name "New name: " dir)))
         (cond ((get-buffer new-name)
                (error "A buffer named '%s' already exists!" new-name))
               (t
```

**Comment by StreakyCobra** [comment]
Looks like an OS X issue then?

**Comment by d12frosted** [comment]
Thanks to @nixmaniack this is fixed in develop.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
